src/org/labkey/test/tests/FolderTest.java

```diff
@@ -255,6 +272,119 @@ public void folderTemplateTest()
         assertElementPresent(Locator.tagContainingText("h3","Child Title"));
     }
 
+    private void createListWithData(String subfolder) throws Exception
+    {
+        _listHelper.createList(getProjectName() + "/" + subfolder, "List1", ListHelper.ListColumnType.AutoInteger, "RowId", new ListHelper.ListColumn("Col1", "ColLabel", ListHelper.ListColumnType.String));
+
+        InsertRowsCommand ir = new InsertRowsCommand("lists", "List1");
+        Map<String, Object> row1 = new HashMap<>();
+        row1.put("Col1", 1);
+
+        ir.addRow(row1);
+
+        Connection cn = new Connection(WebTestHelper.getBaseURL(), PasswordUtil.getUsername(), PasswordUtil.getPassword());
+        ir.execute(cn, getProjectName() + "/" + subfolder);
+    }
+
+    @Test
+    public void deleteFoldersTest() throws Exception
```

Comment:
Split this into multiple `@Test` methods. Delete without children, with children, workbooks via UI, and workbooks via API.
The different scenarios don't share any setup so it wouldn't add much if any overhead.

Reply:
ok, good point.  the last two do share some setup (workbook via UI and workbook via API).  do you feel strongly that these two must be separated?

Reply:
I do not. Three methods seems like a reasonable split.